lib/compute/jenkins-main-node.ts

```diff
@@ -76,6 +82,44 @@ export class JenkinsMainNode {
       }),
     };
 
+    const agentNode = new AgentNode(stack);
+    this.ec2Instance = new Instance(stack, 'MainNode', {
+
+      instanceType: InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE4),
```

Comment:
Why the restructuring?

Reply:
There were a lot of policies that were being created in the constructor and then each of them added individually in a new line. This implementation is much neater and easier to extend in future if we add a new policy. We can now simply go and create a policy in the create policy function and it will be added to the role on ec2